### DISCUSSION TITLE
Nano 33 IoT SAMD21

### DISCUSSION THREAD

**Discussion by nawozef** [hathach/tinyusb#917] [Q&A]
I want to ask you if this library can work with Nano 33 IoT because I tried almost 3 months without any success. I did a WebUSB Interface as a project to obtain some signals with Arduino Leonardo but I bought Arduino 33 IoT to obtain best stability but I didn’t solve how I can change it.
 
When I connect it, usb stack doesn't appear.

Thanks!

**Comment by hathach** [comment]
It works on samd MCU, but for arduino, its core must be ported to support tinyusb, there is a list of core that support tinyusb on arduino  https://github.com/adafruit/Adafruit_TinyUSB_Arduino . If you still want to run tinyusb on nano33, try to ask Arduino to add tinyusb support to their core.

**Reply by nawozef** [reply]
oh thanks!! I don't understand the difference before, i will ask it.

**Reply by hathach** [reply]
You should ask on Arduino forum to add tinyusb as usb stack. Otherwise, your option is using other samd board such as Adafruit.

**Reply by nawozef** [reply]
@hathach Thanks a lot and sorry for open other discussion, at the beggining I understand that I need to ask to this community to add Nano 33 IoT for that reason I create a new discussion which is in category "Ideas", can you please erase it? I'm totally a novice on github asking and also on arduino boards. if they help me to add it, I will share you the answer to attach it as a compatible board.

**Reply by hathach** [reply]
I already made it very clear, you have to do your own research for more information. Sum up, to use tinyusb lib with arduino, you have 2 options:
- either ask https://forum.arduino.cc/ to add support for TinyUSB
- or use of the core that already supported such as Adafruit one.
